core/src/main/java/nl/nn/adapterframework/doc/model/FrankDocModel.java

```diff
@@ -65,6 +63,7 @@
 	private @Getter Map<String, ElementType> allTypes = new HashMap<>();
 	private @Getter Map<ElementRole.Key, ElementRole> allElementRoles = new HashMap<>();
 	private final ElementRole.Factory elementRoleFactory = new ElementRole.Factory();
+	private Map<Set<ElementRole.Key>, ElementRoleSet> allElementRoleSets = new HashMap<>();
```

Comment:
Het FrankDocModel is een soort FrankElementBuilder aan het worden? Misschien is het handig om dit artikel hierover te lezen: https://dzone.com/articles/design-patterns-the-builder-pattern

Reply:
Gelezen. Het is een goed verhaal. Ik heb ook gelezen over een complexer builder pattern waarmee je uit verschillende vormen van invoer objecten kan maken. Dit komt misschien van pas als we de code gaan updaten om de XSDs tijdens de build te genereren.